components/pkg-aci/plan.sh

```diff
@@ -11,7 +12,7 @@ pkg_bin_dirs=(bin)
 program=$pkg_name
 
 do_build() {
-  cp -v $PLAN_CONTEXT/bin/${program}.sh ${program}
+  cp -v "$PLAN_CONTEXT"/bin/${program}.sh ${program}
```

Comment:
I'm a little curious why we're quoting "$PLAN_CONTEXT" instead of quoting this full line and we have curly braces only around $program rather than both things? This feels a bit inconsistent.

Reply:
Your suggestion would work 100%. The reason that it's not *necessary* to quote `${program}` in this case is that shellcheck can see the value it would have here is `hab-pkg-aci` and that wouldn't be re-split or globbed.

What you suggest would be a bit better and more future-proof. I'll make that change, but in general, I tried to make the smallest changes I could to satisfy `shellcheck`, which is why I only added quoting for `$PLAN_CONTEXT`, which `shellcheck` doesn't know enough about to determine whether it needs quoting.

Reply:
Oh, and one other thing. There's no difference between these two:
```bash
"$PLAN_CONTEXT/bin/${program}.sh"
```
```bash
"${PLAN_CONTEXT}/bin/${program}.sh"
```
Unlike double-quotes, braces do not suppress globbing or word-splitting. The only *functional* use they have is to resolve ambiguity about where a token ends. For example:
```bash
$ FOO=a
$ BAR=b
$ FOOBAR=c
$ echo $FOO
a

$ echo $FOOBAR
c

$ echo ${FOO}BAR
aBAR
```
In this case, the token will end with the `/` before `bin`, so there's no ambiguity. I'm torn on the standard of whether we should use braces by default or not. On one hand, they look nice but on the other people who aren't into shell arcana they may be misled to believe they will function similar to double-quoting. In any case our existing code is very inconsistent and currently we have no shell style guide, so maybe that's something we should consider creating one.